pkg/apis/pipeline/v1/param_types_test.go

```diff
@@ -453,3 +454,61 @@ func TestArrayReference(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateParamsTypesInMatrix(t *testing.T) {
+	tests := []struct {
+		name     string
+		matrix   *v1.Matrix
+		wantErrs *apis.FieldError
+	}{{
+		name: "parameters in matrix are strings",
+		matrix: &v1.Matrix{
+			Params: []v1.Param{{
+				Name: "foo", Value: v1.ParamValue{Type: v1.ParamTypeString, StringVal: "foo"},
+			}, {
+				Name: "bar", Value: v1.ParamValue{Type: v1.ParamTypeString, StringVal: "bar"},
+			}}},
+		wantErrs: &apis.FieldError{
+			Message: "invalid value: parameters of type array only are allowed in matrix",
+			Paths:   []string{"matrix[foo]", "matrix[bar]"},
+		},
+	}, {
+		name: "parameters in matrix are arrays",
+		matrix: &v1.Matrix{
+			Params: []v1.Param{{
+				Name: "foobar", Value: v1.ParamValue{Type: v1.ParamTypeArray, ArrayVal: []string{"foo", "bar"}},
+			}, {
+				Name: "barfoo", Value: v1.ParamValue{Type: v1.ParamTypeArray, ArrayVal: []string{"bar", "foo"}},
+			}}},
+	}, {
+		name: "parameters in include matrix are strings",
+		matrix: &v1.Matrix{
+			Include: []v1.MatrixInclude{
+				{Name: "build-1"},
+				{Params: []v1.Param{
+					{Name: "IMAGE", Value: v1.ParamValue{Type: v1.ParamTypeString, StringVal: "image-1"}},
+					{Name: "DOCKERFILE", Value: v1.ParamValue{Type: v1.ParamTypeString, StringVal: "path/to/Dockerfile1"}}}},
+			}},
+	}, {
+		name: "parameters in include matrix are arrays",
+		matrix: &v1.Matrix{
+			Include: []v1.MatrixInclude{
+				{Name: "build-1"},
+				{Params: []v1.Param{
+					{Name: "foobar", Value: v1.ParamValue{Type: v1.ParamTypeArray, ArrayVal: []string{"foo", "bar"}}},
+					{Name: "barfoo", Value: v1.ParamValue{Type: v1.ParamTypeArray, ArrayVal: []string{"bar", "foo"}}}}},
+			}},
+		wantErrs: &apis.FieldError{
+			Message: "invalid value: parameters of type string only are allowed in matrix",
+			Paths:   []string{"matrix[barfoo]", "matrix[foobar]"},
+		},
+	},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			if d := cmp.Diff(tc.wantErrs.Error(), v1.ValidateParamsTypesInMatrix(tc.matrix).Error()); d != "" {
+				t.Errorf("ValidateParamsTypesInMatrix() errors diff %s", diff.PrintWantGot(d))
+			}
+		})
+	}
+}
```

Comment:
we have a test for matrix validations in https://github.com/tektoncd/pipeline/blob/d2f075aba67273ad7f216a6c35160825bcbd035a/pkg/apis/pipeline/v1/pipeline_types_test.go#L592-L702

adds duplication, please add new test cases to the existing validations test